src/components/form/components/numeric_input.tsx

```diff
@@ -0,0 +1,47 @@
+import type { NumericInputProps as BPNumericInputProps } from '@blueprintjs/core';
+import { NumericInput as BPNumericInput } from '@blueprintjs/core';
+
+import { useFieldContext } from '../context/use_ts_form.js';
+import { Label } from '../utils/Label.js';
+import { useErrors } from '../utils/use_errors.js';
+import { useFieldId } from '../utils/use_field_id.js';
+import { useIntent } from '../utils/use_intent.js';
+
+interface NumericInputProps
+  extends Omit<BPNumericInputProps, 'defaultValue' | 'name'> {
+  label?: string;
+  required?: boolean;
+}
+
+export function NumericInput(props: NumericInputProps) {
+  const { label, required, ...rest } = props;
+  const field = useFieldContext<number>();
+  const id = useFieldId(field.name);
+  const error = useErrors(field);
+  const intent = useIntent(error);
+
+  function onChange(value: number) {
+    return field.handleChange(value);
+  }
+
+  return (
+    <Label
```

Comment:
Not sure why you decided to create a separate component for this case but not others.

I would not create simple `FormGroup` wrappers which mostly forward props.